### DISCUSSION TITLE
[Company and projects management] Custom image pull secrets

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#243] [Q&A] [answered]
### Which section?

Console

### Which is the version?

v10.8.0

### Description

Hi guys,
I found [this page](https://docs.mia-platform.eu/docs/development_suite/set-up-infrastructure/create-project#custom-docker-image-pull-imagepullsecret) in the documentation regarding the addition of custom image pull secrets on projects.

Can you confirm that, to add my custom secrets, I need to:
* Create a secret of type `docker-registry` using `mlp` or manually;
* adding the name of the secret inside the `Docker ImagePullSecret names` list at the company or project level.

Thanks.

**Comment by fredmaggiowski** [comment, marked as answer]
Yes, that's exactly what you need to do, remember teo things:

1. you can let mlp create the secret for you (if you already use it)

2. The configuration you provide in the company is inherited by the project only during project creation, so if you need to use this feature in an existing project you'll have to configure the secret name in the project as well

**Reply by frengooo** [reply]
Hi @fredmaggiowski,
thanks for your answer.

1. Yes, I'll let `mlp` create the secret for me of course 😄 
2. Thanks, I will eventually manually propagate the secrets on the existing projects
